packages/compass-saved-aggregations-queries/src/stores/open-item.ts

```diff
@@ -264,28 +404,35 @@ export const openSavedItem =
     }
 
     const { database, collection } = item;
+    const multiConnectionsEnabled =
+      preferencesAccess.getPreferences().enableNewMultipleConnectionSystem;
+
+    if (!multiConnectionsEnabled) {
+      // In single connections mode, we only expect one connections to be here
+      // in the list of active connections.
+      const [activeConnection] = activeConnections;
```

Comment:
Unsure why we need the switch on the feature flag here, this should be the same logic for both multi and single connection mode (with other auto-resolving logic):
```suggestion
    if (activeConnections.length === 1) {
      // If there is only one connection, try to resolve collection name to open the saved item without prompting for input
      const activeConnection = activeConnections[0];
```

Reply:
Yea sorry for misdirection here - after our conversation I totally forgot to update / close this PR and left the work for the new modals and this branching in my other branch. I probably will close this one and will just raise one PR having both the changes.